Clarify the default values for newly created LiveMap and LiveCounter (#478)
ably/specification: specifications/objects-features.md

- `(RTO4)` When a channel `ATTACHED` `ProtocolMessage` is received, the client library must perform the following actions in order. The `ProtocolMessage` may contain a `HAS_OBJECTS` bit flag (see [TR3](../features#TR3)); note that some of the following actions are conditional on this flag.
  - `(RTO4c)` The [RTO17](#RTO17) sync state must transition to `SYNCING` if not already `SYNCING`
  - `(RTO4d)` The `bufferedObjectOperations` list must be cleared without applying any buffered operations
  - `(RTO4a)` If the `HAS_OBJECTS` flag is 1, the server will shortly perform an `OBJECT_SYNC` sequence as described in [RTO5](#RTO5). Note that this does not imply that objects are definitely present on the channel, only that there may be; the `OBJECT_SYNC` message may be empty
  - `(RTO4b)` If the `HAS_OBJECTS` flag is 0 or there is no `flags` field, the sync sequence must be considered complete immediately, and the client library must perform the following actions in order:
    - `(RTO4b1)` All objects except the one with id `root` must be removed from the internal `ObjectsPool`

>     - `(RTO4b2)` The data for the `LiveMap` with id `root` must set to an empty map per [RTLM4c](#RTLM4c). Note that the client SDK must not create a new `LiveMap` instance with id `root`; it must only clear the internal data of the existing `LiveMap` with id `root`

Reviewer: ```suggestion
    - `(RTO4b2)` The data for the `LiveMap` with id `root` must be set to a new map created per [RTLM4c](#RTLM4c). Note that the client SDK must not create a new `LiveMap` instance with id `root`; it must only clear the internal data of the existing `LiveMap` with id `root`
```

Reviewer: Sorry, that first suggestion was wrong, should be 
```suggestion
    - `(RTO4b2)` The data for the `LiveMap` with id `root` must be set to the value described in [RTLM4c](#RTLM4c). Note that the client SDK must not create a new `LiveMap` instance with id `root`; it must only clear the internal data of the existing `LiveMap` with id `root`
```

>     - `(RTO4b2)` The data for the `LiveMap` with id `root` must be cleared by setting it to an empty map per [RTLM4c](#RTLM4c). Note that the client SDK must not create a new `LiveMap` instance with id `root`; it must only clear the internal data of the existing `LiveMap` with id `root`

Reviewer: ```suggestion
    - `(RTO4b2)` The data for the `LiveMap` with id `root` must be cleared by setting it to the value described in [RTLM4c](#RTLM4c). Note that the client SDK must not create a new `LiveMap` instance with id `root`; it must only clear the internal data of the existing `LiveMap` with id `root`
```

Author: We don't really set any value as per `RTLM4c`. Being explicit, we clear/get rid of all values from the map.
Do you think wording gets misleading here?

Reviewer: RTLM4c describes a particular value for `data` (i.e. an empty map); we're saying to set it to that value

Reviewer: again as mentioned in the other comment we can even lose the word "clear"

Author: Addressed, removed `cleared` keyword https://github.com/ably/specification/pull/478/changes/d0de4cab91601e3210be9b8a484ec904494a4ea5

      - `(RTLO4e3a)` This clause has been replaced by [RTLO6a](#RTLO6a)
      - `(RTLO4e3b)` This clause has been replaced by [RTLO6b](#RTLO6b)
        - `(RTLO4e3b1)` This clause has been replaced by [RTLO6b1](#RTLO6b1)

>     - `(RTLO4e4)` Clear the internal data of the `LiveObject` as described in [RTLC4b](#RTLC4b) or [RTLM4c](#RTLM4c), depending on the object type

Reviewer: ```suggestion
    - `(RTLO4e4)` Clear the internal data of the `LiveObject` by setting it to the value described in [RTLC4b](#RTLC4b) or [RTLM4c](#RTLM4c), depending on the object type
```

Author: We don't really set any value for either `LiveMap` or `LiveCounter`, don't you think it's a bit misleading to say `setting it to the value` wdyt?

Reviewer: What do you mean "we don't really set any value" — isn't this spec point saying that you should set the LiveObject's `data` property to a specific value?

Reviewer: It could even just lose the word "clear" and say

```
    - `(RTLO4e4)` Set the `data` attribute of the `LiveObject` to the value described in [RTLC4b](#RTLC4b) or [RTLM4c](#RTLM4c), depending on the object type
```

Author: Updated https://github.com/ably/specification/pull/478/changes/d0de4cab91601e3210be9b8a484ec904494a4ea5
